mlflow/genai/judges/optimizers/dspy.py

```diff
@@ -104,9 +106,12 @@ class CustomPredict(dspy.Predict):
 
             def __init__(self, judge):
                 super().__init__(create_dspy_signature(judge))
-                # Convert MLflow model URI to LiteLLM format for DSPy
-                judge_model_litellm = convert_mlflow_uri_to_litellm(judge.model)
-                self._lm = dspy.LM(model=judge_model_litellm)
+                # Use special AgentEvalLM for databricks model, otherwise convert to LiteLLM
+                if judge.model == _DATABRICKS_DEFAULT_JUDGE_MODEL:
```

Comment:
Let's add a utility method to instantiate a dspy.LM from a `model` parameter, and we can reuse said method here and below.